schema/query/projection_evaluator.go

```diff
@@ -79,33 +90,31 @@ func evalProjection(ctx context.Context, p Projection, payload map[string]interf
 		} else if def != nil {
 			// If field is not found, it may be a connection
 			if ref, ok := def.Validator.(*schema.Connection); ok {
-				// Do not execute the sub-request right away, store a
-				// asyncProjection type of lambda that will be executed later with
-				// concurrency control
-				handler, err := subResourceHandler(pf, def, ref.Path, resolver)
+				q, err := connectionQuery(pf)
 				if err != nil {
 					return nil, err
 				}
-				res[name] = handler
+				rbr.request(ref.Path, q, func(payloads []map[string]interface{}, validator schema.Validator) (err error) {
+					for i := range payloads {
+						if payloads[i], err = evalProjection(ctx, pf.Children, payloads[i], validator, rbr); err != nil {
+							return fmt.Errorf("%s: error applying projection on sub-resource item #%d: %v", pf.Name, i, err)
+						}
+					}
+					resMu.Lock()
```

Comment:
should this be locked _while_ resolveFieldHandler is running, or only when res[name] is set?